tests/types/test_resolver_types.py

```diff
@@ -94,3 +96,12 @@ def get_overlap() -> Union[Venn, Diagram]: ...
 
     resolver = StrawberryResolver(get_overlap)
     assert resolver.type == Union[Venn, Diagram]
+
+
+def test_async_resolver():
```

Comment:
maybe naming it more close to what it is testing?

```suggestion
def test_sync_to_async_resolver():
```